crates/core/src/instance.rs

```diff
@@ -307,9 +307,28 @@ impl Instance {
                             match result {
                                 // If we couldn't run the validation due to unresolved dependencies,
                                 // we have to re-add this entry at the end of the queue:
-                                Err(HolochainError::ValidationPending) => context.block_on(
-                                    queue_holding_workflow(pending.clone(), context.clone()),
-                                ),
+                                Err(HolochainError::ValidationPending) => {
+                                    // And with a delay so we are not trying to re-validate many times per second.
+                                    let mut delay = if let Some(old_delay) = maybe_delay {
+                                        // Exponential back-off:
+                                        // If this was delayed before we double the delay.
+                                        old_delay * 2
+                                    } else {
+                                        Duration::from_secs(10)
+                                    };
```

Comment:
Could also be a one liner let mut delay = maybe_delay.map(|old_delay| old_delay*2).unwrap_or_else(|| Duation::from_secs(10)) but not necessary